Makefile

```diff
@@ -1968,14 +1968,17 @@ ansibleee_cleanup: ## deletes the operator, but does not cleanup the service res
 	${CLEANUP_DIR_CMD} ${OPERATOR_DIR}
 
 ##@ BAREMETAL
+
+BAREMETAL_PREP_DEPS := $(if $(findstring true,$(BMO_SETUP)), crc_bmo_setup)
+
 .PHONY: baremetal_prep
 baremetal_prep: export IMAGE=${BAREMETAL_IMG}
 baremetal_prep: ## creates the files to install the operator using olm
 	$(eval $(call vars,$@,openstack-baremetal))
 	bash scripts/gen-olm.sh
 
 .PHONY: baremetal
-baremetal: operator_namespace baremetal_prep ## installs the operator, also runs the prep step. Set BAREMETAL_IMG for custom image.
+baremetal: operator_namespace $(BAREMETAL_PREP_DEPS) baremetal_prep ## installs the operator, also runs the prep step. Set BAREMETAL_IMG for custom image.
```

Comment:
Thanks! It works in CI prow jobs as multi-node compact cluster has CBO installed. Could have been  `BAREMETAL_DEPS` aligned with the target it's used.